packages/itwin/tree-widget/src/test/trees/category-tree/CategoriesTreeComponent.test.tsx

```diff
@@ -27,31 +40,215 @@ describe("<CategoriesTreeComponent />", () => {
+      label: PropertyRecord.fromString("category-node"),
+      autoExpand: true,
+      hasChildren: false,
+    } as PresentationTreeNodeItem,
+  ];
+
   const filteredCategories: CategoryInfo[] = [{
     categoryId: "FilteredCategoryId",
     subCategoryIds: ["FilteredSubCategoryId1", "FilteredSubCategoryId2"],
   }];
 
-  afterEach(() => {
-    imodelMock.reset();
-    sinon.restore();
-    vpMock = mockViewport();
+  const viewport =  {
+    onDisplayStyleChanged: new BeEvent<(vp: Viewport) => void>(),
+    onViewedCategoriesChanged: new BeEvent<(vp: Viewport) => void>(),
+    view: {
+      is3d: sinon.fake.returns(false),
+    },
+  } as unknown as Viewport;
+
+  const iModel = {
+    createQueryReader: sinon.fake.returns({
+      toArray: sinon.fake.returns(new Promise((resolve) => resolve([]))),
+    }) as unknown as ECSqlReader,
+    categories: {
+      getCategoryInfo: sinon.fake.returns(new Promise((resolve) => resolve(new Map<Id64String, IModelConnection.Categories.CategoryInfo>()))),
+    } as unknown as IModelConnection.Categories,
+  } as unknown as IModelConnection;
+
+  describe("<CategoriesTreeComponent />", () => {
+
+    it("returns null if iModel is undefined", () => {
+      sinon.stub(IModelApp.viewManager, "selectedView").get(() => ({} as Viewport));
+      const result = render(
+        <CategoriesTreeComponent />
+      );
+      expect(result.container.children).to.be.empty;
+    });
+
+    it("returns null if viewport is undefined", () => {
+      sinon.stub(UiFramework, "getIModelConnection").returns({} as IModelConnection);
+      const result = render(
+        <CategoriesTreeComponent />
+      );
+      expect(result.container.children).to.be.empty;
+    });
+
+    it("returns CategoriesTreeComponentImpl when iModel and viewport are defined", async () => {
+      const categoryTreeSpy = sinon.stub(categoryTree, "CategoryTree").returns(<></>);
+      sinon.stub(IModelApp.viewManager, "selectedView").get(() => viewport);
+      sinon.stub(UiFramework, "getIModelConnection").returns(iModel);
+      const result = render(
+        <CategoriesTreeComponent />
+      );
+      expect(result.container.children).to.not.be.empty;
+      expect(categoryTreeSpy).to.be.called;
+    });
+
+    it("getLabel returns translated label of the component", () => {
+      const translateSpy = sinon.stub(TreeWidget, "translate").returns("test categories label");
+      expect(CategoriesTreeComponent.getLabel()).to.be.eq(TreeWidget.translate("test categories label"));
+      expect(translateSpy).to.be.calledWith("categories");
+    });
+
+    describe("getFilteredCategories", () => {
+
+      it("filteredCategories is not undefined when filteredProvider is called", async () => {
+        sinon.stub(IModelApp.viewManager, "selectedView").get(() => viewport);
+        sinon.stub(UiFramework, "getIModelConnection").returns(iModel);
+        sinon.stub(categoryVisibilityHandler, "useCategories").returns([]);
```

Comment:
These stubs are repeated in all tests under `getFilteredCategories`. Maybe move them to `beforeEach` hook under `getFilteredCategories`?

Reply:
fixed